edb/server/protocol/auth_ext/http.py

```diff
@@ -1685,12 +1688,14 @@ def _make_secret_token(
         self,
         identity_id: str,
         secret: str,
+        derive_for_info: str,
         additional_claims: (
             dict[str, str | int | float | bool | None] | None
         ) = None,
         expires_in: datetime.timedelta | None = None,
     ) -> str:
-        signing_key = self._get_auth_signing_key()
+        input_key_material = self._get_auth_signing_key()
+        signing_key = util.derive_key(input_key_material, derive_for_info)
```

Comment:
We talked about putting the derive function on a separate thread but that change is a little non-trivial since most of the call sites like this one are synchronous. Not a huge deal if we still want to do that since the call sites of _these_ functions are usually async, so it would just be one layer of refactoring.

I'll provide some measurements done locally to decide if it's worth doing so now.

Reply:
So, locally, it takes betwen 0.2ms and 0.05ms to run the derive_key function.

Reply:
This seems fast enough, so let's roll with sync generation for now and revisit if this becomes a bottleneck.